File: src/public/ai/explainCard.js
"use strict";

const { requireUser } = require("../../common/auth");

async function handleAiExplainCard({ method, res, auth }) {
  const okRes = requireUser({ auth, res });
  if (okRes !== true) return okRes;

  if (method !== "POST") return res.methodNotAllowed("Method not allowed");
  return res.notImplemented("TODO: AI explain card (public lambda can call model APIs)");
}

module.exports = { handleAiExplainCard };

File: src_C/Public/Ai/ExplainCard.cs
using Amazon.Lambda.APIGatewayEvents;
using RecallSmith.Lambda.Common;

namespace RecallSmith.Lambda.Public.Ai;

public static class ExplainCard
{
  public static Task<APIGatewayProxyResponse> HandleAiExplainCard(LambdaRequest req, Res res, AuthContext auth)
  {
    var deny = Auth.RequireUser(auth, res);
    if (deny is not null) return Task.FromResult(deny);

    if (req.Method != "POST") return Task.FromResult(res.MethodNotAllowed("Method not allowed"));
    return Task.FromResult(res.NotImplemented("TODO: AI explain card (public lambda can call model APIs)"));
  }
}

